cryfs: trigger bottle rebuild

smessmer opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This fixes https://github.com/Homebrew/homebrew-core/issues/35947

freedom1400 commented:
Watsa people

fxcoudert commented:
Can we add a test that looks at what library this is linked against?

MikeMcQuaid commented on Formula/cryfs.rb:
```diff
@@ -47,6 +48,20 @@ def install
+    # the cryfs bottle was compiled with and the crypto++ library installed by homebrew to.
+    Dir.mktmpdir do |basedir|
+      Dir.mktmpdir do |mountdir|
+        assert_match "Mounting filesystem", pipe_output("#{bin}/cryfs -f \"#{basedir}\" \"#{mountdir}\"", "password")
```
- Can use `2>&1` to avoid printing error messages. 
- Can use single quotes to avoid needing to escape the `"`

MikeMcQuaid commented on Formula/cryfs.rb:
```diff
@@ -47,6 +48,20 @@ def install
+        assert_match "Mounting filesystem", pipe_output("#{bin}/cryfs -f \"#{basedir}\" \"#{mountdir}\"", "password")
+      end
+    end
+    print("Note: A permission error in the previous line is expected and still counts as successful test run\n")
```
Can remove this.

smessmer commented on Formula/cryfs.rb:
```diff
@@ -47,6 +48,20 @@ def install
+    # wrong. For example there was an ABI incompatibility issue between the crypto++ version
+    # the cryfs bottle was compiled with and the crypto++ library installed by homebrew to.
+    Dir.mktmpdir do |basedir|
+      Dir.mktmpdir do |mountdir|
```
It could be, but I think `mktmpdir` is better because it automatically cleans up after the block is exited.

smessmer commented:
Thanks. I addressed your comments.

Btw, there was a new upstream release 0.9.10. So instead of bumping the revision number, I switched to the new upstream release.

SMillerDev commented:
@Brewtestbot test this please

smessmer commented:
CI fails with a "directory not empty" error message. Not sure how to fix this, maybe actually a but in homebrew, see https://github.com/Homebrew/brew/issues/4989

smessmer commented:
ok tests seem to have passed now, thanks. Can we merge this?

MikeMcQuaid commented:
@smessmer Can you remove the `mktmpdir` first? Thanks!

MikeMcQuaid commented:
Installation failing on High Sierra with what looks like a Boost issue: https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/37186/version=high_sierra/testReport/junit/brew-test-bot/high_sierra/install_cryfs/

MikeMcQuaid commented:
Reran it and it was fine 🎉.

Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @smessmer!
